app/files/views/user_gallery.py

```diff
@@ -0,0 +1,46 @@
+from rest_framework import status
+from rest_framework.response import Response
+
+from app.common.permissions import BasicViewPermission
+from app.common.viewsets import BaseViewSet
+from app.files.models.user_gallery import UserGallery
+from app.files.serializers.user_gallery import UserGallerySerializer
+
+
+class UserGalleryViewSet(BaseViewSet):
+    serializer_class = UserGallerySerializer
+    permission_classes = [BasicViewPermission]
+    queryset = UserGallery.objects.all()
+
+    def retrieve(self, request, *_args, **_kwargs):
+        """Retrieve all files in gallery"""
+        try:
+            files = UserGallery.get_all_files(request.user)
+            serializer = UserGallerySerializer(files, many=True)
+            return Response(serializer.data, status=status.HTTP_200_OK)
+        except UserGallery.DoesNotExist:
+            return Response(
+                {"detail": "Galleriet finnes ikke"},
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+
+    def create(self, request, *args, **kwargs):
+        """Create a gallery for the current user"""
+        if UserGallery.has_gallery(request.user):
+            return Response(
+                {"detail": "User already has a gallery."},
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+        serializer = UserGallerySerializer(
+            data=request.data, context={"request": request}
+        )
```

Comment:
Siden dere bruke oneToOne field her så er det ikke behov for dette. Det vil bli kastet en feil automatisk hvis en bruekr prøver å lage et nytt et